src/ServiceControl/Monitoring/EndpointMonitoringStats.cs

```diff
@@ -0,0 +1,16 @@
+namespace ServiceControl.Monitoring
+{
+    using System.Threading;
+
+    public class EndpointMonitoringStats
+    {
+        private int active;
+        private int failing;
+
+        public int Active => active;
+        public int Failing => failing;
+
+        public void RecordActive() => Interlocked.Increment(ref active);
```

Comment:
If I read correctly `RecordActive` and `RecordFailing` are only called by  `EndpointInstanceMonitor::AddTo` which seems to be invoked in sync by `GetStats` so I think the `Interlocked` is redundant here.